autodepgraph/graph.py

```diff
@@ -302,6 +317,44 @@ def draw_svg(self, filename: str=None):
         self._update_drawing_attrs()
         vis.draw_graph_svg(self, filename)
 
+    def open_html_viewer(self):
+        """ Open html viewer for the file specified by the svg backend """
+        template = os.path.join(os.path.split(autodepgraph.__file__)[0], 'svg_viewer', 'svg_graph_viewer.html')
+        with open(template, 'rt') as fid:
+            x=fid.read()
+            
+        base, file = os.path.split(self.cfg_svg_filename)        
+        x=x.replace('adg_graph.svg', file)
+        
+        tfile=tempfile.mktemp(prefix='svgviewer-', suffix='html', dir=base)
+        with open(tfile, 'wt') as fid:
+            fid.write(x)
+        webbrowser.open_new_tab(tfile)
+        return tfile
+
+    def set_node_attribute(self, node, attribute, value):
```

Comment:
Good utility addition, currently it is very convoluted to set node attributes. Perhaps we can check for specific attributes (such as the `state`) to ensure that the appropriate set/get methods are called.